File: data/transcoder_evaluation_gfg/go/NUMBER_NON_NEGATIVE_INTEGRAL_SOLUTIONS_B_C_N.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(n int) int {
	var result int = 0
	for i := int(0); i <= n; i++ {
		for j := int(0); j <= n-i; j++ {
			for k := int(0); k <= (n - i - j); k++ {
				if i+j+k == n {
					result++
				}
			}
		}
	}
	return result
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{62, 44, 37, 81, 14, 20, 76, 72, 96, 52}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i]) == f_gold(param0[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/NUMBER_NON_NEGATIVE_INTEGRAL_SOLUTIONS_B_C_N.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut n: i32) -> i32 {
    let mut result: i32 = 0 as i32;
    let mut i: i32 = 0 as i32;
    while i <= n {
        let mut j: i32 = 0 as i32;
        while j <= n - i {
            let mut k: i32 = 0 as i32;
            while k <= n - i - j { if i + j + k == n { result += 1 } k += 1 }
            j += 1
        }
        i += 1
    }
    return result;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [62 as i32, 44 as i32, 37 as i32,
         81 as i32, 14 as i32, 20 as i32,
         76 as i32, 72 as i32, 96 as i32,
         52 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize]) == f_gold(param0[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
